### PR TITLE
Shadowdom v0 and v1

### PR DESCRIPTION
**Pull request by tomalec**
Use Shadow DOM v1 if available.

Support both.
Support even composition with mixed syntax in both environments.

Related
https://github.com/Starcounter/starcounter-include/issues/19
https://github.com/Starcounter/RebelsLounge/issues/72

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -139,6 +142,18 @@ Will make slot names look like this
...
 We will append new `<slot>` (`<content>`) element(-s) to the end of the composition.
 
+
+### Force Shadow DOM v0
+If for any reason you would like to force this element to use Shadow DOM v0,
+you can do that by setting the flag **before** you import this element:
+
+```html
+<script>
+    window.JuicyComposition.shadow = 'v0';
```
Comment (alshakero) [inline README.md]: This will throw `Can't set property 'shadow' of undefined` error. Right?

**Review by alshakero — CHANGES_REQUESTED** [review]
Looks great. Just the forced shadow v0 comment. And I think a spec is needed for that. Apprently the URL query one is passing. But `window.JuicyComposition.shadow` isn't. As per my manual tests.

File: README.md (hunk 1 above)
Comment (tomalec) [reply README.md]: fair point

**Comment by tomalec** [comment]
I made it mostly tu run tests.

>  And I think a spec is needed for that.

All the `?juicy-composition-shadow=v0` tests are the ones that tests it.

Does `window.JuicyComposition={shadow: 'v0'}` works for you?

**Review by alshakero — APPROVED** [review]

**Comment by alshakero** [comment]
@tomalec yes, it's working now!

**Pull request merged**